Graphics/VulkanEngine.h

```diff
@@ -30,14 +72,70 @@ namespace leap::graphics
 		VkPhysicalDevice m_PhysicalDevice{ VK_NULL_HANDLE }; // Vulkan physical device handle
 		VkDevice m_Device{ VK_NULL_HANDLE }; // Vulkan logical device handle
 		VkSurfaceKHR m_Surface{ VK_NULL_HANDLE }; // Vulkan surface handle
+		void InitializeVulkan();
 
 		// Swap chain
 		VkSwapchainKHR m_SwapChain{ VK_NULL_HANDLE };
 		VkFormat m_SwapChainImageFormat{ VK_FORMAT_UNDEFINED };
 		std::vector<VkImage> m_SwapChainImages;
 		std::vector<VkImageView> m_SwapChainImageViews;
 
-		void InitializeVulkan();
 		void InitializeSwapChain();
+
+		// Queues & buffers
+		VkQueue m_GraphicsQueue{ VK_NULL_HANDLE };
+		uint32_t m_GraphicsQueueFamily{ UINT32_MAX };
+		VkCommandPool m_CommandPool{ VK_NULL_HANDLE };
+		VkCommandBuffer m_MainCommandBuffer{ VK_NULL_HANDLE };
+		void InitCommands();
+
+		// Render passes
+		VkRenderPass m_RenderPass{ VK_NULL_HANDLE };
+		std::vector<VkFramebuffer> m_Framebuffers;
+		void InitializeDefaultRenderPass();
+		void InitializeFramebuffers();
+
+		// Render loop
+		VkSemaphore m_PresentSemaphore{ VK_NULL_HANDLE };
+		VkSemaphore m_RenderSemaphore{ VK_NULL_HANDLE };
+		VkFence m_RenderFence{ VK_NULL_HANDLE };
+		void InitializeSyncStructures();
+
+		// Shaders
+		bool LoadShaderModule(const char* filePath, VkShaderModule* outShaderModule);
+		void InitializePipelines();
+
+		// Graphics pipeline
+		VkPipelineLayout m_MeshPipelineLayout{ VK_NULL_HANDLE };
+		VkPipeline m_MeshPipeline{ VK_NULL_HANDLE };
+
+		// Deletion
+		DeletionQueue m_MainDeletionQueue;
+
+		// Allocator
+		VmaAllocator m_Allocator{ VK_NULL_HANDLE };
+
+		// Mesh
+		Mesh m_TeapotMesh;
```

Comment:
Temp?

Reply:
Is indeed temp and only used during development of the core features. This helps me rappidly test and diagnose new feature implementations. Should get removed and reworked into the main engine once everything is setup.